aries_cloudagent/revocation/models/revocation_registry.py

```diff
@@ -169,23 +170,33 @@ async def retrieve_tails(self):
             tails_file_dir.mkdir(parents=True)
 
         buffer_size = 65536  # should be multiple of 32 bytes for sha256
+        file_hasher = hashlib.sha256()
         with open(tails_file_path, "wb", buffer_size) as tails_file:
-            file_hasher = hashlib.sha256()
-
             with Session() as req_session:
                 try:
                     resp = req_session.get(self._tails_public_uri, stream=True)
+                    # Should this directly raise an Error?
+                    if resp.status_code != http.HTTPStatus.OK:
+                        LOGGER.warning(
+                            f"Unexpected status code for tails file: {resp.status_code}"
+                        )
                     for buf in resp.iter_content(chunk_size=buffer_size):
                         tails_file.write(buf)
                         file_hasher.update(buf)
                 except RequestException as rx:
                     raise RevocationError(f"Error retrieving tails file: {rx}")
 
-            download_tails_hash = base58.b58encode(file_hasher.digest()).decode("utf-8")
-            if download_tails_hash != self.tails_hash:
-                raise RevocationError(
-                    "The hash of the downloaded tails file does not match."
-                )
+        download_tails_hash = base58.b58encode(file_hasher.digest()).decode("utf-8")
+        if download_tails_hash != self.tails_hash:
+            try:
+                os.remove(tails_file_path)
+                tails_file_dir.rmdir()
+            except OSError:
+                LOGGER.warning(f"Could not delete invalid tails file: {OSError}")
```

Comment:
OSError is a class instance, that's probably not going to print the desired result. You might want something like `except OSError as err:`